src/main/java/dev/ebyrdeu/movie_jakarta/resource/MovieResource.java

```diff
@@ -0,0 +1,80 @@
+package dev.ebyrdeu.movie_jakarta.resource;
+
+import dev.ebyrdeu.movie_jakarta.dto.MovieDto;
+import dev.ebyrdeu.movie_jakarta.dto.Movies;
+import dev.ebyrdeu.movie_jakarta.entity.Movie;
+import dev.ebyrdeu.movie_jakarta.repository.MovieRepository;
+import jakarta.inject.Inject;
+import jakarta.validation.constraints.NotNull;
+import jakarta.ws.rs.*;
+import jakarta.ws.rs.core.EntityTag;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+
+import java.net.URI;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@Path("/movies")
+public class MovieResource {
+
+    private MovieRepository movieRepository;
+
+    public MovieResource() {
+    }
+
+    @Inject
+    public MovieResource(MovieRepository movieRepository) {
+        this.movieRepository = movieRepository;
+    }
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public Movies all() {
+        return new Movies(movieRepository.findAll().stream().map(MovieDto::map).collect(Collectors.toList()));
+    }
+
+    @GET
+    @Path("/{id}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response one(@PathParam("id") @NotNull UUID id) {
+        Optional<Movie> optionalMovie = movieRepository.findById(id);
+        if (optionalMovie.isPresent()) {
+            MovieDto movieDto = MovieDto.map(optionalMovie.get());
+            return Response.ok(movieDto).build();
+        } else {
+            return Response.status(Response.Status.NOT_FOUND).build();
+        }
+    }
+
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response add(@NotNull MovieDto movieDto) {
+        Movie movie = movieRepository.saveMovie(MovieDto.map(movieDto));
+        return Response.created(URI.create("http://localhost:8080/app/api/movies/" + movie.getId())).build();
+    }
+
+    @PUT
+    @Path("/{id}")
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response update(@PathParam("id") @NotNull UUID id, @NotNull MovieDto movieDetails) {
+        try{
+            movieRepository.updateMovie(id, MovieDto.map(movieDetails));
+        }catch (NotFoundException e){
+            return Response.status(Response.Status.NOT_FOUND).build();
+        }
+        return Response.ok().build();
```

Comment:
```suggestion
   movieRepository.updateMovie(id, MovieDto.map(movieDetails));
        return Response.ok().build();
```

Reply:
That's actually a good point and it would indeed work. The main issue for me was because we already use a `NotFoundException` when calling both the update and delete methods in the Repository layer. This seemed redundant, and I initially thought something wasn't functioning correctly, prompting you to implement an additional layer of protection.

Reply:
I was just focused on catching the thrown exception from the called methods. But when the repository throws an exception it still returns the correct response. Should we remove the try-catch? The only thing is that it catches the exception and respond with the same response code that would be returned without the try-catch.

Reply:
Huh, does it? That feels wrong. It should not... That what I have in my app.
![image](https://github.com/Ebyrdeu/movie_jakarta/assets/42122071/c867ecdd-950f-42d5-b52e-9dee35097e17)